pkg/automationconfig/automation_config_builder.go

```diff
@@ -191,3 +191,90 @@ func withFCV(fcv string) func(*Process) {
 		process.FeatureCompatibilityVersion = fcv
 	}
 }
+
+// CustomRoleBuilder builds a custom role.
+type CustomRoleBuilder struct {
+	role                       string
+	db                         string
+	privileges                 []Privilege
+	roles                      []Role
+	authenticationRestrictions []AuthenticationRestriction
+}
+
+// NewCustomRoleBuilder creates a new CustomRoleBuilder.
+func NewCustomRoleBuilder() *CustomRoleBuilder {
+	return &CustomRoleBuilder{
+		privileges: []Privilege{},
+		roles:      []Role{},
+	}
+}
+
+// WithRole sets the name of a custom role.
+func (b *CustomRoleBuilder) WithRole(role string) *CustomRoleBuilder {
+	b.role = role
+	return b
+}
+
+// WithDB sets the database of a custom role.
+func (b *CustomRoleBuilder) WithDB(db string) *CustomRoleBuilder {
+	b.db = db
+	return b
+}
+
+// AddPrivilege adds a privilege to a custom role.
+func (b *CustomRoleBuilder) AddPrivilege(
+	db string,
+	collection string,
+	cluster bool,
+	anyResource bool,
+	actions []string) *CustomRoleBuilder {
+
+	res := Resource{
+		DB:         db,
+		Collection: collection,
+	}
+	if cluster {
+		res = Resource{
```

Comment:
we lose the values for `DB` and `Collection` with this re-assignment right?

Can we just do 

```golang
res := Resource{
		DB:         db,
		Collection: collection,
                Cluster: cluster,
                AnyResource: anyResource,
	}
```

Reply:
My intention when writing that was to omit the fields (since the accepted format does not allow db/collection to coexist with either of the boolean flags). However, with the string references and the refactoring to use a converter this works. Done!